gadgets/top_file/program.bpf.c

```diff
@@ -0,0 +1,139 @@
+/* SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause) */
+/* Copyright (c) 2021 Hengqi Chen */
+/* Copyright (c) 2023-2024 The Inspektor Gadget authors */
+
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+
+#include <gadget/mntns_filter.h>
+#include <gadget/types.h>
+#include <gadget/macros.h>
+
+#include "stat.h"
+
+#define PATH_MAX 4096
+#define TASK_COMM_LEN 16
+#define TYPE_LEN 1
+
+enum op {
+	READ,
+	WRITE,
+};
+
+struct file_id {
+	__u64 inode;
+	__u32 dev;
+	__u32 pid;
+	__u32 tid;
+};
+
+struct file_stat {
+	gadget_mntns_id mntns_id;
+	__u64 reads;
+	__u64 rbytes;
+	__u64 writes;
+	__u64 wbytes;
+	__u32 pid;
+	__u32 tid;
+	__u8 file[PATH_MAX];
+	__u8 comm[TASK_COMM_LEN];
+	// file type: R, S, O. Using byte array to force parsing as string.
+	__u8 t[TYPE_LEN];
```

Comment:
Do you mean to use an array of chars?

Reply:
It will be parsed as a number by the columns library.

Reply:
@flyth any comment here?

Reply:
> @flyth any comment here?

I think the important part is that it is an array of length 1 rather than a single `__u8` - the latter would just be read as a number, while the former should be interpreted as []byte (at least for now) anyway. Regarding `__u8` vs `unsigned char`, I don't really have a strong opinion.

Reply:
What about using an enumeration for this?

Reply:
Using enum.